test/extended/util/docker.go

```diff
@@ -30,7 +37,55 @@ func PullImage(name string) error {
 		Repository: name,
 		Tag:        "latest",
 	}
-	return client.PullImage(opts, dockerClient.AuthConfiguration{})
+	return client.PullImage(opts, authCfg)
+}
+
+//PushImage, as the name implies, initiates the equivalent of a `docker push` for the "name" parameter to the local registry
+func PushImage(name string, authCfg dockerClient.AuthConfiguration) error {
+	client, err := tutil.NewDockerClient()
+	if err != nil {
+		return err
+	}
+	opts := dockerClient.PushImageOptions{
+		Name: name,
+		Tag:  "latest",
+	}
+	return client.PushImage(opts, authCfg)
+}
+
+//ConstructAuthCfg constructs a non-standard dockerClient.AuthConfiguration that can be used to communicate with the openshift internal docker registry
+func ConstructAuthCfg(oc *CLI) (*dockerClient.AuthConfiguration, error) {
+	authCfg := &dockerClient.AuthConfiguration{}
+	secrets, err := oc.Run("get").Args("secrets", "-o", "json", "--config", KubeConfigPath()).Output()
+	secretList := &kapi.SecretList{}
+	err = gson.Unmarshal([]byte(secrets), secretList)
+	g.By(fmt.Sprintf("\n\n  json unmarsall err %v \n\n", err))
+	if err == nil {
+		for _, secret := range secretList.Items {
+			g.By(fmt.Sprintf("\n\n  secret name %s \n\n", secret.ObjectMeta.Name))
+			if strings.Contains(secret.ObjectMeta.Name, "builder-dockercfg") {
+				dockercfgToken := secret.Data[".dockercfg"]
+				dockercfgTokenJson := string(dockercfgToken)
+				g.By(fmt.Sprintf("\n\n  docker cfg token json %s \n\n", dockercfgTokenJson))
+
+				creds := credentialprovider.DockerConfig{}
+				err = gson.Unmarshal(dockercfgToken, &creds)
+				g.By(fmt.Sprintf("\n\n  json unmarshall err %v \n\n", err))
+				if err == nil {
+					for key, dce := range creds {
```

Comment:
why a for loop if you're always going to break after processing the first item?

(and are we confident that there's only ever one item in the map and it's the correct one?)